### DISCUSSION TITLE
Model Viewer Camera Question

### DISCUSSION THREAD

**Discussion by emincanyuksel** [google/model-viewer#4533] [Q&A]
Hello everyone, I want to look at the model in first person. Is this possible with WASD keys instead of the mouse?

**Comment by elalish** [comment]
For accessibility the camera is already controllable with the arrow keys and page-up, page-down. You can also add eventlisteners for other keys to create whatever camera behaviors you desire.

**Reply by emincanyuksel** [reply]
I understand that, but is it possible to walk around the model as if it were the character I want?

**Reply by elalish** [reply]
Sure, but you'll have to write that code in your event listeners 😄

**Reply by emincanyuksel** [reply]
I do not understand how that 😄

**Comment by vincenttschanz** [comment]
I would like to revive this question. How complex would it be to add the following features to Model Viewer ?

- First person walking mode (like in a Doom like game) including :
- WASD controls for walking in the scene constrained to an horizontal plane
- mouse used to look around
- PgUp/PgDn (or c/v) to move upward/downward
-  no wall detection needed, it's possible to walk through the walls and objects.

I am not a dev, but if someone would be capable of adding theses features I would be happy to pay for that work.
I'am looking for a simple way to present my creations in Blender in a html page. So my worklow would be export to gltf > load it in model viewer inside a html page > publish it.

**Reply by elalish** [reply]
No need to add it to model-viewer - you can add this easily in your own JS with a few event listeners. If you really don't want to try it yourself, I bet an AI could manage it for you.
